docs/data/data-grid/editing/EditingWithDatePickers.tsx

```diff
@@ -0,0 +1,343 @@
+      showTime ? value.getHours() : 0,
+      showTime ? value.getMinutes() : 0,
+      0,
+      0,
+    );
+    const cellValueMs = dateCopy.getTime();
+
+    return compareFn(cellValueMs, filterValueMs);
+  };
+}
+
+function getDateFilterOperators(
+  showTime: boolean = false,
+): GridColTypeDef['filterOperators'] {
+  const InputComponent = showTime ? GridFilterDateTimeInput : GridFilterDateInput;
+
+  return [
+    {
+      value: 'is',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 === value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'not',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 !== value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'after',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 > value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'onOrAfter',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 >= value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'before',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 < value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'onOrBefore',
+      getApplyFilterFn: (filterItem) => {
+        return buildApplyDateFilterFn(
+          filterItem,
+          (value1, value2) => value1 <= value2,
+          showTime,
+        );
+      },
+      InputComponent,
+    },
+    {
+      value: 'isEmpty',
+      getApplyFilterFn: () => {
+        return ({ value }): boolean => {
+          return value == null;
+        };
+      },
+    },
+    {
+      value: 'isNotEmpty',
+      getApplyFilterFn: () => {
+        return ({ value }): boolean => {
+          return value != null;
+        };
+      },
+    },
+  ];
+}
+
+const dateAdapter = new AdapterDateFns({ locale });
+
+/**
+ * `date` column
+ */
+
+const dateColumnType: GridColTypeDef = {
+  ...GRID_DATE_COL_DEF,
+  resizable: false,
+  renderEditCell: (params) => {
+    return <GridEditDateCell {...params} />;
+  },
+  filterOperators: getDateFilterOperators(),
+  valueFormatter: (params) => {
+    if (params.value) {
+      return dateAdapter.format(params.value, 'keyboardDate');
+    }
+    return '';
+  },
+};
+
+function GridEditDateCell({
+  id,
+  field,
+  value,
+}: GridRenderEditCellParams<Date | null>) {
+  const apiRef = useGridApiContext();
+
+  function handleChange(newValue: unknown) {
```

Comment:
We usually use arrow functions inside components. Did you copy this from the docs?

```suggestion
  const handleChange = (newValue: unknown) => {
```

Reply:
No, I just don't have strong preference here